web_src/js/components/ContextPopup.vue

```diff
@@ -30,20 +30,14 @@ import {SvgIcon} from '../svg.js';
 const {appSubUrl, i18n} = window.config;
 
 // NOTE: see models/issue_label.go for similar implementation
-const srgbToLinear = (color) => {
-  color /= 255;
-  if (color <= 0.04045) {
-    return color / 12.92;
-  }
-  return ((color + 0.055) / 1.055) ** 2.4;
-};
-const luminance = (colorString) => {
-  const r = srgbToLinear(parseInt(colorString.substring(0, 2), 16));
-  const g = srgbToLinear(parseInt(colorString.substring(2, 4), 16));
-  const b = srgbToLinear(parseInt(colorString.substring(4, 6), 16));
-  return 0.2126 * r + 0.7152 * g + 0.0722 * b;
+const useLightTextColor = (label) => {
+  // sRGB color space luminance
+  const r = parseInt(label.color.substring(0, 2), 16);
+  const g = parseInt(label.color.substring(2, 4), 16);
+  const b = parseInt(label.color.substring(4, 6), 16);
+  const luminance = (0.299 * r + 0.587 * g + 0.114 * b) / 255;
+  return luminance < 0.35;
```

Comment:
I don't think we can share the implementation between Go and Javascript. To solve the duplication I guess we'd have to render the contents of this popup server side, which seems like a big change.

Reply:
I think the origin question is about `ContextPopup.vue` and `repo-projects.js`, they are bot JS related.

Since `ContextPopup.vue` shouldn't import `repo-projects.js`, so maybe the common code could be moved into `utils.js`, then make `ContextPopup.vue` and  `repo-projects.js` use the common function from `utils.js`